Resources/Computing/ARCComputingElement.py

```diff
@@ -76,7 +76,8 @@ def __writeXRSL( self, executableFile ):
 
   def _reset( self ):
     self.queue = self.ceParameters['Queue']
-    self.gridEnv = self.ceParameters['GridEnv']
+    if 'GridEnv' in self.ceParameters:
+      self.gridEnv = self.ceParameters['GridEnv']
```

Comment:
You can do
self.gridEnv = self.ceParameters.get('GridEnv')

This will set self.gridEnv to None if it is not in the dictionary